lib/spack/spack/compilers/__init__.py

```diff
@@ -628,7 +659,7 @@ def arguments_to_detect_version_fn(operating_system, paths):
     def _default(search_paths):
         command_arguments = []
         files_to_be_tested = fs.files_in(*search_paths)
-        for compiler_name in spack.compilers.supported_compilers():
+        for compiler_name in spack.compilers.supported_compilers_for_host_platform():
```

Comment:
(no change needed) I was mildly concerned that this might complicate cross compiling cases, but our compiler objects don't differentiate between host and target platform, so that would have to be resolved later anyway.